### DISCUSSION TITLE
Does Qwen (DashScope) only support cloud service calls? Can the locally private deployed Qwen large model support calls as well?

### DISCUSSION THREAD

**Discussion by shaojool** [langchain4j/langchain4j#2173] [Q&A]

**Comment by jiangsier-xyz** [comment]
The langchain4j-dashscope is a wrapper for the model service provider DashScope (by Aliyun). If you want to use some local runtime framework to run the Qwen2 model and implement the relevant interfaces of langchain4j, you will need to implement it separately.

**Comment by sakoojiang** [comment]
you can use ollama to use local qwen2 or qwen2.5 model

1. install ollama  
2. ollama pull  qwen2:1.5b
3. use ollama model
```
ChatLanguageModel model = OllamaChatModel.builder()
            .baseUrl("http://localhost:11434")
            .modelName("qwen2:1.5b")
            .build();
```
